### PR TITLE
mon: bail from handle_command() if _generate_command_map() fails

### PR DESCRIPTION
**Pull request by nkshirsagar**
Bail out of handle_command() if _generate_command_map() throws an exception due to invalid json. This is an additional fix for the one that went in through
1109e62f6a3c0a7e06a088c4b833c8ecec51f791

Fixes: https://tracker.ceph.com/issues/57859
Signed-off-by: Nikhil Kshirsagar <nikhil.kshirsagar@canonical.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [x] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by nkshirsagar** [comment]
jenkins test make check

File: src/mon/Monitor.cc [hunk 1]
```diff
@@ -3445,6 +3445,7 @@ void Monitor::handle_command(MonOpRequestRef op)
   }
   catch(bad_cmd_get& e) {
```
Comment (idryomov) [inline src/mon/Monitor.cc:3446]: I'd also fix the formatting and missing const while at it:
```suggestion
  } catch (const bad_cmd_get& e) {
```

**Review by idryomov — COMMENTED** [review]
> Fixes: https://tracker.ceph.com/issues/54558

I'd generate a new ticket, linked to the above, because this fixup needs to be backported to pacific and quincy.

Also, I'd like to draw @ceph/core reviewer's attention to a potential wider breakage in this area: https://github.com/ceph/ceph/pull/45547#pullrequestreview-1099607585.

File: src/mon/Monitor.cc (hunk 1 above)
Comment (nkshirsagar) [reply src/mon/Monitor.cc]: Hi @idryomov , thanks!

I created https://tracker.ceph.com/issues/57859 for this, and re-submitted the MR. 

Please may I request you to link this pull request in it, I am not able to modify the description once I created the issue.

Regards,
Nikhil.

**Comment by nkshirsagar** [comment]
jenkins retest this please

File: src/mon/Monitor.cc (hunk 1 above)
Comment (idryomov) [reply src/mon/Monitor.cc]: > Please may I request you to link this pull request in it, I am not able to modify the description once I created the issue.

Done

**Comment by idryomov** [comment]
Please change the title of the commit to match the title of the PR.  Otherwise LGTM!

**Comment by nkshirsagar** [comment]
> Please change the title of the commit to match the title of the PR. Otherwise LGTM!

done! thanks!

**Review by idryomov — APPROVED** [review]

**Review by rzarzynski — APPROVED** [review]

**Comment by nkshirsagar** [comment]
jenkins retest this please

**Comment by ljflores** [comment]
Rados suite review: http://pulpito.front.sepia.ceph.com/?branch=wip-yuri7-testing-2022-10-17-0814

Failures:, unrelated
    1. https://tracker.ceph.com/issues/57311
    2. https://tracker.ceph.com/issues/52321
    3. https://tracker.ceph.com/issues/52657
    4. https://tracker.ceph.com/issues/57935

Details:
    1. rook: ensure CRDs are installed first - Ceph - Orchestrator
    2. qa/tasks/rook times out: 'check osd count' reached maximum tries (90) after waiting for 900 seconds - Ceph - Orchestrator
    2. rados/thrash-erasure-code: wait_for_recovery timeout due to "active+clean+remapped+laggy" pgs - Ceph - RADOS
    3. MOSDPGLog::encode_payload(uint64_t): Assertion `HAVE_FEATURE(features, SERVER_NAUTILUS)' - Ceph - RADOS
    4. all test jobs get stuck at "Running task ansible.cephlab..." - Infrstructure - Sepia

**Pull request merged**